### DISCUSSION TITLE
ROS2 Driver connection failure

### DISCUSSION THREAD

**Discussion by anu120** [UniversalRobots/Universal_Robots_ROS2_Driver#1194] [Q&A]
Hello. I am following the given steps on github and UR documentation (h[ttps://docs.universal-robots.com/Universal_Robots_ROS2_Documentation/index.html#](url)). But the driver does not get activated properly. Subnet and default gateways of both remote PC and robot are matched. the Ping command is also a success. I could gather the calibration file also from the robot. but when trying to launch the driver , The teaching pendant of the robot says 

> connection to the remote PC could not be establishes. Reason: connection refused

 Below is the error I get on remote PC when trying to launch the driver using the prompt '''ros2 launch ur_robot_driver ur_control.launch.py ur_type:=ur3 robot_ip:=<> launch_rviz:=true
'''.  

Error:

> INFO] [launch]: Default logging verbosity is set to INFO
> [INFO] [dashboard_client-2]: process started with pid [19429]
> [INFO] [controller_stopper_node-3]: process started with pid [19431]
> [INFO] [ur_ros2_control_node-1]: process started with pid [19427]
> [INFO] [urscript_interface-4]: process started with pid [19433]
> [INFO] [robot_state_publisher-5]: process started with pid [19435]
> [INFO] [rviz2-6]: process started with pid [19437]
> [INFO] [spawner-7]: process started with pid [19439]
> [INFO] [spawner-8]: process started with pid [19441]
> [INFO] [spawner-9]: process started with pid [19443]
> [controller_stopper_node-3] [INFO] [1732748172.525638671] [Controller stopper]: Waiting for switch controller service to come up on controller_manager/switch_controller
> [ur_ros2_control_node-1] [WARN] [1732748172.526785355] [controller_manager]: [Deprecated] Passing the robot description parameter directly to the control_manager node is deprecated. Use '~/robot_description' topic from 'robot_state_publisher' instead.
> [ur_ros2_control_node-1] text not specified in the tf_prefix tag
> [ur_ros2_control_node-1] [INFO] [1732748172.527107483] [resource_manager]: Loading hardware 'ur3' 
> [ur_ros2_control_node-1] [INFO] [1732748172.528526055] [resource_manager]: Initialize hardware 'ur3' 
> [ur_ros2_control_node-1] [INFO] [1732748172.528578343] [resource_manager]: Successful initialization of hardware 'ur3'
> [ur_ros2_control_node-1] [INFO] [1732748172.528754345] [resource_manager]: 'configure' hardware 'ur3' 
> [ur_ros2_control_node-1] [INFO] [1732748172.528761173] [URPositionHardwareInterface]: Starting ...please wait...
> [ur_ros2_control_node-1] [INFO] [1732748172.528772750] [URPositionHardwareInterface]: Initializing driver...
> [ur_ros2_control_node-1] [WARN] [1732748172.529339619] [UR_Client_Library:]: Your system/user seems not to be setup for FIFO scheduling. We recommend using a lowlatency kernel with FIFO scheduling. See https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/blob/master/ur_robot_driver/doc/real_time.md for details.
> [dashboard_client-2] [INFO] [1732748172.530038053] [UR_Client_Library:]: Connected: Universal Robots Dashboard Server
> [dashboard_client-2] 
> [ur_ros2_control_node-1] [INFO] [1732748172.530566654] [UR_Client_Library:]: Negotiated RTDE protocol version to 2.
> [ur_ros2_control_node-1] [INFO] [1732748172.530738266] [UR_Client_Library:]: Setting up RTDE communication with frequency 125.000000
> [robot_state_publisher-5] [INFO] [1732748172.534985768] [robot_state_publisher]: got segment base
> [robot_state_publisher-5] [INFO] [1732748172.535044446] [robot_state_publisher]: got segment base_link
> [robot_state_publisher-5] [INFO] [1732748172.535049505] [robot_state_publisher]: got segment base_link_inertia
> [robot_state_publisher-5] [INFO] [1732748172.535052621] [robot_state_publisher]: got segment flange
> [robot_state_publisher-5] [INFO] [1732748172.535055545] [robot_state_publisher]: got segment forearm_link
> [robot_state_publisher-5] [INFO] [1732748172.535058273] [robot_state_publisher]: got segment ft_frame
> [robot_state_publisher-5] [INFO] [1732748172.535060877] [robot_state_publisher]: got segment shoulder_link
> [robot_state_publisher-5] [INFO] [1732748172.535063556] [robot_state_publisher]: got segment tool0
> [robot_state_publisher-5] [INFO] [1732748172.535066107] [robot_state_publisher]: got segment upper_arm_link
> [robot_state_publisher-5] [INFO] [1732748172.535068715] [robot_state_publisher]: got segment world
> [robot_state_publisher-5] [INFO] [1732748172.535071250] [robot_state_publisher]: got segment wrist_1_link
> [robot_state_publisher-5] [INFO] [1732748172.535073716] [robot_state_publisher]: got segment wrist_2_link
> [robot_state_publisher-5] [INFO] [1732748172.535076117] [robot_state_publisher]: got segment wrist_3_link
> [spawner-7] [INFO] [1732748172.766364226] [spawner_scaled_joint_trajectory_controller]: waiting for service /controller_manager/list_controllers to become available...
> [spawner-9] [INFO] [1732748172.791923171] [spawner_forward_position_controller]: waiting for service /controller_manager/list_controllers to become available...
> [spawner-8] [INFO] [1732748172.793007576] [spawner_joint_state_broadcaster]: waiting for service /controller_manager/list_controllers to become available...
> [rviz2-6] [INFO] [1732748172.809579528] [rviz2]: Stereo is NOT SUPPORTED
> [rviz2-6] [INFO] [1732748172.809649738] [rviz2]: OpenGl version: 4.6 (GLSL 4.6)
> [rviz2-6] [INFO] [1732748172.821236352] [rviz2]: Stereo is NOT SUPPORTED
> [ur_ros2_control_node-1] [FATAL] [1732748173.569255200] [URPositionHardwareInterface]: Variable 'speed_slider_mask' is currently controlled by another RTDE client. The input recipe can't be used as configured
> [ur_ros2_control_node-1] [INFO] [1732748173.569350485] [resource_manager]: Failed to 'configure' hardware 'ur3'
> [controller_stopper_node-3] [INFO] [1732748173.575323946] [Controller stopper]: Service available
> [controller_stopper_node-3] [INFO] [1732748173.575431900] [Controller stopper]: Waiting for list controllers service to come up on controller_manager/list_controllers
> [controller_stopper_node-3] [INFO] [1732748173.575532361] [Controller stopper]: Service available
> [ur_ros2_control_node-1] [WARN] [1732748173.579962832] [controller_manager]: Could not enable FIFO RT scheduling policy
> [ur_ros2_control_node-1] [INFO] [1732748173.773382649] [controller_manager]: Loading controller 'scaled_joint_trajectory_controller'
> [ur_ros2_control_node-1] [INFO] [1732748173.786431047] [scaled_joint_trajectory_controller]: Using scaling state from the hardware from interface speed_scaling/speed_scaling_factor.
> [ur_ros2_control_node-1] [WARN] [1732748173.788488556] [scaled_joint_trajectory_controller]: [Deprecated]: "allow_nonzero_velocity_at_trajectory_end" is set to true. The default behavior will change to false.
> [spawner-7] [INFO] [1732748173.798079067] [spawner_scaled_joint_trajectory_controller]: Loaded scaled_joint_trajectory_controller
> [ur_ros2_control_node-1] [INFO] [1732748173.798764383] [controller_manager]: Loading controller 'forward_position_controller'
> [ur_ros2_control_node-1] [INFO] [1732748173.812468628] [controller_manager]: Loading controller 'joint_state_broadcaster'
> [spawner-9] [INFO] [1732748173.813436747] [spawner_forward_position_controller]: Loaded forward_position_controller
> [ur_ros2_control_node-1] [INFO] [1732748173.828481921] [controller_manager]: Configuring controller 'scaled_joint_trajectory_controller'
> [ur_ros2_control_node-1] [INFO] [1732748173.828867837] [scaled_joint_trajectory_controller]: No specific joint names are used for command interfaces. Using 'joints' parameter.
> [ur_ros2_control_node-1] [INFO] [1732748173.828901875] [scaled_joint_trajectory_controller]: Command interfaces are [position] and state interfaces are [position velocity].
> [ur_ros2_control_node-1] [INFO] [1732748173.828949975] [scaled_joint_trajectory_controller]: Using 'splines' interpolation method.
> [spawner-8] [INFO] [1732748173.830227279] [spawner_joint_state_broadcaster]: Loaded joint_state_broadcaster
> [ur_ros2_control_node-1] [INFO] [1732748173.831695224] [scaled_joint_trajectory_controller]: Controller state will be published at 100.00 Hz.
> [ur_ros2_control_node-1] [INFO] [1732748173.836727021] [scaled_joint_trajectory_controller]: Action status changes will be monitored at 20.00 Hz.
> [ur_ros2_control_node-1] [INFO] [1732748173.844608112] [controller_manager]: Configuring controller 'forward_position_controller'
> [ur_ros2_control_node-1] [INFO] [1732748173.845801399] [forward_position_controller]: configure successful
> [ur_ros2_control_node-1] [INFO] [1732748173.852430779] [controller_manager]: Configuring controller 'joint_state_broadcaster'
> [ur_ros2_control_node-1] [INFO] [1732748173.852547392] [joint_state_broadcaster]: 'joints' or 'interfaces' parameter is empty. All available state interfaces will be published
> [ur_ros2_control_node-1] [ERROR] [1732748173.860513968] [resource_manager]: Not acceptable command interfaces combination: 
> [ur_ros2_control_node-1] Start interfaces: 
> [ur_ros2_control_node-1] [
> [ur_ros2_control_node-1]   shoulder_pan_joint/position
> [ur_ros2_control_node-1]   shoulder_lift_joint/position
> [ur_ros2_control_node-1]   elbow_joint/position
> [ur_ros2_control_node-1]   wrist_1_joint/position
> [ur_ros2_control_node-1]   wrist_2_joint/position
> [ur_ros2_control_node-1]   wrist_3_joint/position
> [ur_ros2_control_node-1] ]
> [ur_ros2_control_node-1] Stop interfaces: 
> [ur_ros2_control_node-1] [
> [ur_ros2_control_node-1] ]
> [ur_ros2_control_node-1] Not available: 
> [ur_ros2_control_node-1] [
> [ur_ros2_control_node-1]  shoulder_pan_joint/position
> [ur_ros2_control_node-1]  shoulder_lift_joint/position
> [ur_ros2_control_node-1]  elbow_joint/position
> [ur_ros2_control_node-1]  wrist_1_joint/position
> [ur_ros2_control_node-1]  wrist_2_joint/position
> [ur_ros2_control_node-1]  wrist_3_joint/position
> [ur_ros2_control_node-1] ]
> [ur_ros2_control_node-1] 
> [ur_ros2_control_node-1] [ERROR] [1732748173.860555618] [controller_manager]: Could not switch controllers since prepare command mode switch was rejected.
> [spawner-7] [ERROR] [1732748173.861770301] [spawner_scaled_joint_trajectory_controller]: Failed to activate controller
> [ur_ros2_control_node-1] [ERROR] [1732748173.868186266] [joint_state_broadcaster]: None of requested interfaces exist. Controller will not run.
> [ur_ros2_control_node-1] [WARN] [1732748173.868290893] [joint_state_broadcaster]: Error occurred while doing error handling.
> [ur_ros2_control_node-1] [ERROR] [1732748173.868340527] [controller_manager]: After activation, controller 'joint_state_broadcaster' is in state 'unconfigured' (1), expected 'active' (3).
> [spawner-8] [INFO] [1732748173.876901466] [spawner_joint_state_broadcaster]: Configured and activated joint_state_broadcaster
> [ur_ros2_control_node-1] [INFO] [1732748173.878210023] [controller_manager]: Loading controller 'io_and_status_controller'
> [spawner-8] [INFO] [1732748173.884808181] [spawner_joint_state_broadcaster]: Loaded io_and_status_controller
> [ur_ros2_control_node-1] [INFO] [1732748173.885351332] [controller_manager]: Configuring controller 'io_and_status_controller'
> [ur_ros2_control_node-1] [ERROR] [1732748173.893084504] [resource_manager]: Not acceptable command interfaces combination: 
> [ur_ros2_control_node-1] Start interfaces: 
> [ur_ros2_control_node-1] [
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_0
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_1
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_2
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_3
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_4
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_5
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_6
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_7
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_8
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_9
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_10
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_11
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_12
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_13
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_14
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_15
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_16
> [ur_ros2_control_node-1]   gpio/standard_digital_output_cmd_17
> [ur_ros2_control_node-1]   gpio/standard_analog_output_cmd_0
> [ur_ros2_control_node-1]   gpio/standard_analog_output_cmd_1
> [ur_ros2_control_node-1]   gpio/tool_voltage_cmd
> [ur_ros2_control_node-1]   gpio/io_async_success
> [ur_ros2_control_node-1]   speed_scaling/target_speed_fraction_cmd
> [ur_ros2_control_node-1]   speed_scaling/target_speed_fraction_async_success
> [ur_ros2_control_node-1]   resend_robot_program/resend_robot_program_cmd
> [ur_ros2_control_node-1]   resend_robot_program/resend_robot_program_async_success
> [ur_ros2_control_node-1]   payload/mass
> [ur_ros2_control_node-1]   payload/cog.x
> [ur_ros2_control_node-1]   payload/cog.y
> [ur_ros2_control_node-1]   payload/cog.z
> [ur_ros2_control_node-1]   payload/payload_async_success
> [ur_ros2_control_node-1]   zero_ftsensor/zero_ftsensor_cmd
> [ur_ros2_control_node-1]   zero_ftsensor/zero_ftsensor_async_success
> [ur_ros2_control_node-1]   hand_back_control/hand_back_control_cmd
> [ur_ros2_control_node-1]   hand_back_control/hand_back_control_async_success
> [ur_ros2_control_node-1]   gpio/analog_output_domain_cmd
> [ur_ros2_control_node-1] ]
> [ur_ros2_control_node-1] Stop interfaces: 
> [ur_ros2_control_node-1] [
> [ur_ros2_control_node-1] ]
> [ur_ros2_control_node-1] Not available: 
> [ur_ros2_control_node-1] [
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_0
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_1
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_2
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_3
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_4
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_5
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_6
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_7
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_8
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_9
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_10
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_11
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_12
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_13
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_14
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_15
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_16
> [ur_ros2_control_node-1]  gpio/standard_digital_output_cmd_17
> [ur_ros2_control_node-1]  gpio/standard_analog_output_cmd_0
> [ur_ros2_control_node-1]  gpio/standard_analog_output_cmd_1
> [ur_ros2_control_node-1]  gpio/tool_voltage_cmd
> [ur_ros2_control_node-1]  gpio/io_async_success
> [ur_ros2_control_node-1]  speed_scaling/target_speed_fraction_cmd
> [ur_ros2_control_node-1]  speed_scaling/target_speed_fraction_async_success
> [ur_ros2_control_node-1]  resend_robot_program/resend_robot_program_cmd
> [ur_ros2_control_node-1]  resend_robot_program/resend_robot_program_async_success
> [ur_ros2_control_node-1]  payload/mass
> [ur_ros2_control_node-1]  payload/cog.x
> [ur_ros2_control_node-1]  payload/cog.y
> [ur_ros2_control_node-1]  payload/cog.z
> [ur_ros2_control_node-1]  payload/payload_async_success
> [ur_ros2_control_node-1]  zero_ftsensor/zero_ftsensor_cmd
> [ur_ros2_control_node-1]  zero_ftsensor/zero_ftsensor_async_success
> [ur_ros2_control_node-1]  hand_back_control/hand_back_control_cmd
> [ur_ros2_control_node-1]  hand_back_control/hand_back_control_async_success
> [ur_ros2_control_node-1]  gpio/analog_output_domain_cmd
> [ur_ros2_control_node-1] ]
> [ur_ros2_control_node-1] 
> [ur_ros2_control_node-1] [ERROR] [1732748173.893140900] [controller_manager]: Could not switch controllers since prepare command mode switch was rejected.
> [spawner-8] [ERROR] [1732748173.893645218] [spawner_joint_state_broadcaster]: Failed to activate controller
> [ERROR] [spawner-7]: process has died [pid 19439, exit code 1, cmd '/opt/ros/humble/lib/controller_manager/spawner scaled_joint_trajectory_controller -c /controller_manager --controller-manager-timeout 10 --ros-args'].
> [INFO] [spawner-9]: process has finished cleanly [pid 19443]
> [ERROR] [spawner-8]: process has died [pid 19441, exit code 1, cmd '/opt/ros/humble/lib/controller_manager/spawner --controller-manager /controller_manager --controller-manager-timeout 10 joint_state_broadcaster io_and_status_controller speed_scaling_state_broadcaster force_torque_sensor_broadcaster tcp_pose_broadcaster ur_configuration_controller --ros-args'].

There are no other remote PCs connected to the UR3, just this active PC and its teaching pendant. Highly appreciate if you can assist me to find a solution for this error. Thank you!

**Comment by urfeex** [comment]
The culprit can probably be found in your output:

```
[ur_ros2_control_node-1] [FATAL] [1732748173.569255200] [URPositionHardwareInterface]: Variable 'speed_slider_mask' is currently controlled by another RTDE client. The input recipe can't be used as configured
```

This usually happens when a fieldbus (Either Ethernet/IP or Profinet) is enabled. See https://docs.universal-robots.com/Universal_Robots_ROS2_Documentation/doc/ur_robot_driver/ur_robot_driver/doc/usage/toc.html#when-starting-the-driver-it-crashes-with-variable-speed-slider-mask-is-currently-controlled-by-another-rtde-client

**Reply by anu120** [reply]
Thank you very much. It resolved the driver activation error. Now I can visualize the movement of real robot on RViz. However, when I try to use External Control in the program tree of the Teaching Pendant (TP), the TP still shows below error 

> connection to the remote PC could not be established. Reason: connection refused

Does this mean that the connection is one way only? How can I go about this to resolve?

**Reply by urfeex** [reply]
That can be another issue, which I at first was thinking: There might be a firewall active on your control PC. The robot will want to connect to ports 50001-50004 unless manually changed. Those have to be opened in case there's a firewall active.

**Reply by anu120** [reply]
Thank you, however the firewall is disabled already. Driver connection error is also not completely gone. Sometimes it connects and sometimes its not. unable to figure out the reason.

**Comment by marols27** [comment]
Hello, we got a similar problem using a ur5e, folowing the instructions at: https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver.

The computer system is a ubuntu VM using VM-ware (player 16) using windows 10 and 11 as hosts (we are 2 people collaborating).
We are able to ping the robot from the VM, so we assume the networking is setup correctly. 

We are using the prompt: ros2 launch ur_robot_driver ur_control.launch.py ur_type:=ur5e robot_ip:= 172.31.1.144


```
[INFO] [launch]: All log files can be found below /home/prosjekt/.ros/log/2025-03-12-15-22-10-883220-ROS2-jazzy-jalisco-318102
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [dashboard_client-2]: process started with pid [318120]
[INFO] [controller_stopper_node-3]: process started with pid [318121]
[INFO] [ros2_control_node-1]: process started with pid [318119]
[INFO] [urscript_interface-4]: process started with pid [318122]
[INFO] [robot_state_publisher-5]: process started with pid [318123]
[INFO] [rviz2-6]: process started with pid [318124]
[INFO] [spawner-7]: process started with pid [318125]
[INFO] [spawner-8]: process started with pid [318126]
[controller_stopper_node-3] [INFO] [1741789331.557913141] [Controller stopper]: Waiting for switch controller service to come up on controller_manager/switch_controller
[robot_state_publisher-5] [INFO] [1741789331.775876782] [robot_state_publisher]: Robot initialized
[dashboard_client-2] [INFO] [1741789331.870086355] [UR_Client_Library:]: Connected: Universal Robots Dashboard Server
[dashboard_client-2] 
[ros2_control_node-1] [INFO] [1741789331.915801683] [controller_manager]: Subscribing to '/robot_description' topic for robot description.
[ros2_control_node-1] [INFO] [1741789331.917017232] [controller_manager]: update rate is 500 Hz
[ros2_control_node-1] [INFO] [1741789331.917029175] [controller_manager]: Spawning controller_manager RT thread with scheduler priority: 50
[ros2_control_node-1] [WARN] [1741789331.917359164] [controller_manager]: Could not enable FIFO RT scheduling policy: with error number <1>(Operation not permitted). See [https://control.ros.org/master/doc/ros2_control/controller_manager/doc/userdoc.html] for details on how to enable realtime scheduling.
[ros2_control_node-1] [INFO] [1741789331.934038308] [controller_manager]: Received robot description from topic.
[ros2_control_node-1] text not specified in the tf_prefix tag
[ros2_control_node-1] [INFO] [1741789331.950969098] [controller_manager.resource_manager]: Loading hardware 'ur5e' 
[ros2_control_node-1] [INFO] [1741789331.968873254] [controller_manager.resource_manager]: Loaded hardware 'ur5e' from plugin 'ur_robot_driver/URPositionHardwareInterface'
[ros2_control_node-1] [INFO] [1741789331.968998070] [controller_manager.resource_manager]: Initialize hardware 'ur5e' 
[ros2_control_node-1] [INFO] [1741789331.969182298] [controller_manager.resource_manager]: Successful initialization of hardware 'ur5e'
[ros2_control_node-1] [INFO] [1741789331.974429857] [resource_manager]: 'configure' hardware 'ur5e' 
[ros2_control_node-1] [INFO] [1741789331.974511362] [URPositionHardwareInterface]: Starting ...please wait...
[ros2_control_node-1] [INFO] [1741789331.974531660] [URPositionHardwareInterface]: Initializing driver...
[ros2_control_node-1] [WARN] [1741789332.020568076] [UR_Client_Library:]: Your system/user seems not to be setup for FIFO scheduling. We recommend using a lowlatency kernel with FIFO scheduling. See https://docs.universal-robots.com/Universal_Robots_ROS_Documentation/doc/ur_client_library/doc/real_time.html for details.
[ros2_control_node-1] [INFO] [1741789332.277800543] [UR_Client_Library:]: Negotiated RTDE protocol version to 2.
[ros2_control_node-1] [INFO] [1741789332.278663395] [UR_Client_Library:]: Setting up RTDE communication with frequency 500.000000
[rviz2-6] [INFO] [1741789332.667587987] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-6] [INFO] [1741789332.667795690] [rviz2]: OpenGl version: 4.5 (GLSL 4.5)
[rviz2-6] [INFO] [1741789332.837005320] [rviz2]: Stereo is NOT SUPPORTED
[ros2_control_node-1] [INFO] [1741789333.322072015] [URPositionHardwareInterface]: Calibration checksum: 'calib_12788084448423163542'.
[spawner-8] [INFO] [1741789333.595712427] [spawner_joint_trajectory_controller]: waiting for service /controller_manager/list_controllers to become available...
[spawner-7] [INFO] [1741789333.631271237] [spawner_joint_state_broadcaster]: waiting for service /controller_manager/list_controllers to become available...
[ros2_control_node-1] [ERROR] [1741789334.337255646] [URPositionHardwareInterface]: The calibration parameters of the connected robot don't match the ones from the given kinematics config file. Please be aware that this can lead to critical inaccuracies of tcp positions. Use the ur_calibration tool to extract the correct calibration from the robot and pass that into the description. See [https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/blob/main/ur_calibration/README.md] for details.
[ros2_control_node-1] [INFO] [1741789334.337354553] [URPositionHardwareInterface]: System successfully started!
[ERROR] [ros2_control_node-1]: process has died [pid 318119, exit code -11, cmd '/opt/ros/jazzy/lib/controller_manager/ros2_control_node --ros-args --params-file /home/prosjekt/ros2_ws/install/ur_robot_driver/share/ur_robot_driver/config/ur5e_update_rate.yaml --params-file /tmp/launch_params_ya4pkx24'].
[spawner-8] [FATAL] [1741789343.689063474] [spawner_joint_trajectory_controller]: Could not contact service /controller_manager/list_controllers
[spawner-7] [FATAL] [1741789343.707990514] [spawner_joint_state_broadcaster]: Could not contact service /controller_manager/list_controllers
[ERROR] [spawner-8]: process has died [pid 318126, exit code 1, cmd '/opt/ros/jazzy/lib/controller_manager/spawner --controller-manager /controller_manager --controller-manager-timeout 10 --inactive joint_trajectory_controller forward_velocity_controller forward_position_controller force_mode_controller passthrough_trajectory_controller freedrive_mode_controller --ros-args'].
[ERROR] [spawner-7]: process has died [pid 318125, exit code 1, cmd '/opt/ros/jazzy/lib/controller_manager/spawner --controller-manager /controller_manager --controller-manager-timeout 10 joint_state_broadcaster io_and_status_controller speed_scaling_state_broadcaster force_torque_sensor_broadcaster tcp_pose_broadcaster ur_configuration_controller scaled_joint_trajectory_controller --ros-args'].
```

**Reply by urfeex** [reply]
@marols27 I cannot really see an overlap with the initial discussion as in your case the driver seems to be crashing. Could you please create an issue explaining in more detail all the steps that you are doing? As a side note: We do not recommend using a VM for running the driver as that might have poor real-time and network performance compared to a native / dockerized system.

**Reply by marols27** [reply]
Will do
